File: server/models/transactionModel.js
import mongoose from "mongoose";

const transactionSchema = mongoose.Schema({
  transactionID: {
    type: String,
    required: true,
  },
  sender: {
    type: String,
    required: true,
  },
  senderName: {
    type: String,
    required: true,
  },
  receiver: {
    type: String,
    required: true,
  },
  receiverName: {
    type: String,
    required: true,
  },
  amount: {
    type: Number,
    required: true,
  },
  dateAndTime: {
    type: String,
    required: true,
  },
});

const Transaction = mongoose.model("Transaction", transactionSchema);

export default Transaction;

File: lib/models/transaction_model.dart
// ignore_for_file: public_member_api_docs, sort_constructors_first
import 'dart:convert';

class Transaction {
  final String id;
  final String transactionID;
  final String sender;
  final String senderName;
  final String receiver;
  final String receiverName;
  final double amount;
  final dynamic dateAndTime;

  Transaction({
    required this.id,
    required this.transactionID,
    required this.sender,
    required this.senderName,
    required this.receiver,
    required this.receiverName,
    required this.amount,
    this.dateAndTime,
  });

  Map<String, dynamic> toMap() {
    return {
      'id': id,
      'transactionID': transactionID,
      'sender': sender,
      'senderName': senderName,
      'receiver': receiver,
      'receiverName': receiverName,
      'amount': amount,
      'dateAndTime': dateAndTime,
    };
  }

  factory Transaction.fromMap(Map<String, dynamic> map) {
    return Transaction(
      id: map['_id'] ?? '',
      transactionID: map['transactionID'] ?? '',
      sender: map['sender'] ?? '',
      senderName: map['senderName'] ?? '',
      receiver: map['receiver'] ?? '',
      receiverName: map['receiverName'] ?? '',
      amount: map['amount']?.toDouble() ?? 0.0,
      dateAndTime: map['dateAndTime'] ?? '',
    );
  }

  String toJson() => json.encode(toMap());

  factory Transaction.fromJson(String source) =>
      Transaction.fromMap(json.decode(source) as Map<String, dynamic>);
}
